### DISCUSSION TITLE
Multi tenant subdomains (Check it out to know how !)

### DISCUSSION THREAD

**Discussion by opestro** [coollabsio/coolify#1948] [General]
Hi , i just started using coolify , i like it is so simple and has a lot of features. I want to ask about the wildcard domain for one app, i mean i have a multi tenant app like a saas so users can access the dashboard by their [id].domain.ltd for example , so the id is dynamic. I read the documents . You said that i need to enter the subdomain but what about the dynamic subdomain ?

**Comment by grahamhealy20** [comment]
Maybe this will help? https://github.com/coollabsio/coolify/discussions/1905

**Comment by mrvfino** [comment]
@opestro any luck on this? I'm trying to figure out this as well :)

**Reply by opestro** [reply]
Yes, i fix it, and it's working for me . Now it's 1:28 AM , tomorrow ill send you how i make it work

**Reply by mrvfino** [reply]
Thank you!

**Reply by opestro** [reply]
Hi, i'm back :
post this lines on your container labels where the project is :
traefik.http.routers.https-0-<PROJECT-NAME>.rule=Host(`<DOMAIN-NAME>`) && PathPrefix(`/`) || HostRegexp(`{subdomain:[a-zA-Z0-9-]+}.<DOMAIN-NAME>`)
Must looks like this one :
traefik.http.routers.https-0-yggckok.rule=Host(`khfif.shop`) && PathPrefix(`/`) || HostRegexp(`{subdomain:[a-zA-Z0-9-]+}.khfif.shop`)
![Capture d'écran 2024-06-16 090321](https://github.com/coollabsio/coolify/assets/96887412/d13bda03-c985-4de2-937f-32620e0614d7)

**Reply by mrvfino** [reply]
Thanks, trying this out!

**Reply by mrvfino** [reply]
Hi @opestro , checked it after the fix for container labels in the latest versions. Unfortunately still doesn'tw rok for me.

May i know if you also implemented a `dnsChallenge` on your traefik server?
Also, i noticed that you're using `http` does it work for you in `https` as well?

Thanks.

Edit: Nvm lol. I finally found where the logs are shown (i kept digging on my hetzner account), as it turns out I just had to correct the missing single quotes on my traefik config.

Works on https also:
https://test.pricekodo.com/
https://123.pricekodo.com/

Also as it turns out there's no need to configure the `dnsChallenge` on the server's proxy.

**Reply by benediktpannermayr** [reply]
Hi,
this configuration works great.
Unfortunately, I don't get letsencrypt certificates automatically assigned.

@mrvfino How did you solve that?

Best regards
Benedikt Pannermayr

**Reply by mrvfino** [reply]
@benediktpannermayr , not sure about the letsencrypt part. There's no need to do anything on the proxy. Just copy the container labels shown above but make sure you're using the unique suffix. Hope this helps :)

**Reply by benediktpannermayr** [reply]
Hi @mrvfino,
thank you for your quick and helpful answer.

Unfortunately I still dont't get any certificates assigned.

I'll take a closer look at this in the next few days.

Best regards
Benedikt Pannermayr

**Reply by mrvfino** [reply]
Hi @benediktpannermayr,

I have some update check. I did another deployment for a different application of mine that uses subdomains and I also encountered a certificate issue. Wherein only a default Traefik certificate is assigned to pages that use subdomains. 

For some reason following this: https://coolify.io/docs/knowledge-base/traefik/wildcard-certificates helped. So I retract what I said about not doing anything with regards to the proxy.

The odd thing is that, I reverted my proxy configuration back and the certificate fix lingered, that's why I thought that it had nothing to do with wildcard subdomains (in https), but it actually does.

**Reply by yewyewxd** [reply]
@mrvfino  Do you set `https://pricekodo.com` as the "Wildcard Domain" for your server? In Servers -> localhost -> General -> "Wildcard Domain" input field

**Reply by mrvfino** [reply]
@yewyewXD hmmm I don't remember doing that. Just checked, that I actually didn't.

The key part for me was following the instruction for wildcard certs and updating my proxy configuration for the server.

**Reply by allisonverdam** [reply]
> @yewyewXD hmmm I don't remember doing that. Just checked, that I actually didn't.
> 
> The key part for me was following the instruction for wildcard certs and updating my proxy configuration for the server.
> 
> 

What do you do in the server proxy?

**Reply by mrvfino** [reply]
> > @yewyewXD hmmm I don't remember doing that. Just checked, that I actually didn't.
> > The key part for me was following the instruction for wildcard certs and updating my proxy configuration for the server.
> 
> What do you do in the server proxy?

I haven't been using coolify for a long time, so I can't completely remember it. But what I'm sure about is you can start here:
https://coolify.io/docs/knowledge-base/proxy/traefik/wildcard-certs

That setting should apply to the general settings of your coolify instance. Not at a project level.
